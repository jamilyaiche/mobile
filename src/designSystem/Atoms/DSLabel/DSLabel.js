import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'
import Style from '../../../assets/styles/Style'

const DSLabel = (props) => {
  return (

        <Text
            allowFontScaling={props.allowFontScaling}
            {...(props.onLayout !== undefined && { onLayout: props.onLayout })}
            {...(props.onLongPress !== undefined && { onLongPress: props.onLongPress })}
            {...(props.onPress !== undefined && { onPress: props.onPress })}
            {...(props.onTextLayout !== undefined && { onTextLayout: props.onTextLayout })}
            selectable={props.selectable}
            style={props.style}
            {...props}
        >
            {props.children}
        </Text>
  )
}

DSLabel.defaultProps = {
  allowFontScaling: false,
  selectable: false,
  style: Style.TextInputLabelStyle1
}

DSLabel.propTypes = {
  allowFontScaling: PropTypes.bool,
  onLayout: PropTypes.func,
  onLongPress: PropTypes.func,
  onPress: PropTypes.func,
  onTextLayout: PropTypes.func,
  selectable: PropTypes.bool
}

export default DSLabel
