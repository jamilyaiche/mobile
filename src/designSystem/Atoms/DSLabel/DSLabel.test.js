import React from 'react';
import renderer from 'react-test-renderer'
import DSLabel from "./DSLabel"

describe('DSLabel tests',()=>{
    it('text Label Snapshot',()=>{
        const wrapper=renderer.create(<DSLabel>test</DSLabel>)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
})