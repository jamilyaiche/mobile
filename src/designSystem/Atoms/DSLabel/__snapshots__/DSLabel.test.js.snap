// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSLabel tests text Label Snapshot 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "color": undefined,
      "fontFamily": "Roboto-Bold",
      "fontSize": 12,
      "fontWeight": "500",
      "height": 14,
      "left": 12,
      "letterSpacing": 1,
      "lineHeight": 14,
      "paddingHorizontal": 8,
      "position": "absolute",
      "top": -5,
    }
  }
>
  test
</Text>
`;
